src/libsyntax/ext/tt/quoted.rs

```diff
@@ -289,14 +289,14 @@ where
             // `tree` is followed by an `ident`. This could be `$meta_var` or the `$crate` special
             // metavariable that names the crate of the invokation.
             Some(tokenstream::TokenTree::Token(ident_span, ref token)) if token.is_ident() => {
-                let ident = token.ident().unwrap();
+                let (ident, _) = token.ident().unwrap();
                 let span = ident_span.with_lo(span.lo());
                 if ident.name == keywords::Crate.name() {
```

Comment:
Hmm, I think this will accept `$r#crate` in the crate root meaning while we should treat it as a usual metavar named "crate".

Reply:
Could you add a test case for this as well?

Reply:
Currently `r#crate` is forbidden along with the rest of the keywords matching `is_path_segment_keyword`.